compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1419,7 +1419,69 @@ OMR::ResolvedMethodSymbol::getNumberOfBackEdges()
    return numBackEdges;
    }
 
+/*
+ * Returns the bytecode info for an OSR point, taking into account the
+ * possibility that the treetop for a call has different BCI to the OSR
+ * point.
+ */
+TR_ByteCodeInfo
+OMR::ResolvedMethodSymbol::getOSRByteCodeInfo(TR::Node *node)
+   {
+   TR_ByteCodeInfo nodeBCI = node->getByteCodeInfo();
+   if (node->getNumChildren() > 0 && (node->getOpCodeValue() == TR::treetop || node->getOpCode().isCheck()))
+      nodeBCI = node->getFirstChild()->getByteCodeInfo();
+   return nodeBCI;
+   }
+
+/*
+ * Checks the provided node is pending push store or load expected to be
+ * around an OSR point.
+ */
+bool
+OMR::ResolvedMethodSymbol::isOSRRelatedNode(TR::Node *node)
+   {
+   return (node->getOpCode().isStoreDirect()
+         && node->getOpCode().hasSymbolReference()
+         && node->getSymbolReference()->getSymbol()->isPendingPush())
+      || (node->getOpCodeValue() == TR::treetop
```

Comment:
What is the logic behind anchored loads - do we get this structure as a result of the way trees are generated? There is normally no guarantee a load is 'anchored' so a comment on the logic surrounding this check would be good.